### DISCUSSION TITLE
searXNG / upgrade + search engine stats

### DISCUSSION THREAD

**Discussion by un99known99** [searxng/searxng#2845] [Q&A] [answered]
Hi,

just switched from searX to searXNG as I read that searX is no longer maintained. So far I really like the searXNG, setup stuff was very similar to searX :)

I do have 2 questions:

* how is searXNG upgraded (I installed via step-by-step installation (https://docs.searxng.org/admin/installation-searxng.html)?
* the search stats collected are being purged after restarting searXNG (running it via systemd) - can it be kept permanent?

Thx

**Comment by unixfox** [comment]
I think with the step by step installation you need to do `git pull` for updating then restart searxng.

The stats should be permanent if you have redis configured, otherwise they are not.

**Comment by return42** [comment, marked as answer]
Sorry for my late response .. 

> The stats should be permanent if you have redis configured, otherwise they are not.

We plan to store the stats in redis .. but its not implemented yet .. as you have noticed, currently stats get lost after restart.

> how is searXNG upgraded 

See https://docs.searxng.org/admin/update-searxng.html

The script::

    sudo -H ./utils/searxng.sh instance update

was intended for the [Installation Script](https://docs.searxng.org/admin/installation-scripts.html#installation-scripts)prozedure, but should also work when you installed "Step by Step".  The steps to update are shown here in the script:

https://github.com/searxng/searxng/blob/1f835a9f1f6b444bbe7f3b82349444cb050db9c0/utils/searxng.sh#L565-L588

more or less its the same what @unixfox wrote above ..

> I think with the step by step installation you need to do `git pull` for updating then restart searxng.

Git pull does not work, when the branch has been rebased, thats why the script uses:

     git fetch origin "$GIT_BRANCH" 
     git reset --hard "origin/$GIT_BRANCH"

**Comment by un99known99** [comment]
many thanks for the answers

**Comment by un99known99** [comment]
@return42 
Hi, just wanted to give feedback, thx for the information with the upgrade script, it ran fine, updated to 2023.10.1 - I guess due to "rolling release" this should be last official commit?

I faced some "error" during upgrade about not installed / not implemented apache/nginx/uWSGI/pkg_install LXC_BASE_PACKAGES, this I would likely see as a "warn", the red "error" give a bad feeling :)

Despite that I guess that's it with upgrade task(s)?
